code/iomethods.py

```diff
@@ -518,5 +544,132 @@ def decode_time_from_string(timeString):
+    longitudeVariable.long_name = "Longitude"
+
+    latitudeVariable = ncdf.createVariable("latitude", "double", dimensions=("latitude",))
+    latitudeVariable.units = "degrees_north"
+    latitudeVariable.long_name = "Latitude"
+
+    timeVariable = ncdf.createVariable("time", "double", dimensions=("time",))
+
+    ch4Variable = ncdf.createVariable("ch4", np.float, dimensions=("time","latitude","longitude",))
+    ch4Variable.comments = "Unknown1 variable comment"
+    ch4Variable.long_name = "IR BT (add 75 to this value)"
+    ch4Variable.units = ""
+    ch4Variable.grid_name = "grid01"
+    ch4Variable.grid_type = "linear"
+    ch4Variable.level_description = "Earth surface"
+    ch4Variable.time_statistic = "instantaneous"
+    ch4Variable.missing_value = float(330)
+
+    mergFileName = fileName  # Parse file name to get date and time to set as an attribute of timeVariable
+    year = mergFileName[5:9]    # Fix this later to use the functions in iomethods
+    month = mergFileName[9:11]
+    day = mergFileName[11:13]
+    hour = mergFileName[13:15]
+
+    timeVariable.units = "hours since " + year + "-" + month + "-" + day + " " + hour
+
+    longitudeVariable[:] = lon  # Assign numPy arrays to netCDF variables
+    latitudeVariable[:] = lat
+    ch4Variable[:,:,:] = t
+
+    ncdf.close()
+
+
+def read_MERG_pixel_file(path, shape=(2, 3298, 9896), offset=75.):
+    '''
+    Read MERG brightness temperature from binary file. Thanks to Brian Wilson for this contribution.
+    File contains two large arrays (2 time epochs: on the hour and the half hour)
+    of temperature (Kelvin) as an unsigned integer byte, offset by 75 so it will fit in the 0-255 range.
+    For documentation, see http://www.cpc.ncep.noaa.gov/products/global_precip/html/README.
+    '''
+    f = open(path, 'rb')
+    x = np.fromfile(f, dtype=np.uint8, count=-1)       # count=-1 means read entire file
+    f.close()
+
+    t = x.astype(np.float).reshape(shape)
+    t += offset
+
+    lon = np.arange(0.0182, 360., 0.036378335, dtype=np.float)
+
+    lat = np.arange(59.982, -60., -0.036383683, dtype=np.float)
+
+    return lon, lat, t
+
+# **********************************************************************************************************************
+
+def read_binary_data(userVariables):
+    '''
+
+    '''
+
+    global LAT
+    global LON
+
+    userVariables.filelist = get_fileList_for_binaries(userVariables.DIRS['MERGBinaryDirName'],
+                                                       userVariables.startDateTime, userVariables.endDateTime)
+
+    userVariables.filelist.sort()
+
+    inputList = []
+    timeList = []
+
+    nfiles = len(userVariables.filelist)
+
+    # Crash if there are no binary files
+    if nfiles == 0:
+        print 'Error: No files in this directory! Exiting elegantly'
+        sys.exit()
+
+    for files in userVariables.filelist:
+        try:
+                bLON, bLAT, t = read_MERG_pixel_file(files)
```

Comment:
@caocampb Lines 653 - 658 are indented 8 spaces instead of 4.  Try blocks are indented the same amount as any loop or conditional statement.